packages/core/src/utils/viaProxy.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Utility to redirect remote sources via Proxy
+ *
+ * This utility is used to rewrite a request to be routed through
+ * the Proxy instead.
+ */
+
+/**
+ * Virtual Module provided by the Motion Canvas
+ * Plugin. Returns `false` if proxy is set to `false`, else `true`
+ */
+//@ts-ignore
+import {enabled} from 'virtual:proxysettings';
```

Comment:
I'm not sure how I feel about using virtual modules in `core`.
So far, all integration with Vite was done through `import.meta` so it was possible to use `core` like a normal typescript library and build it without Vite. But with virtual modules, one will need to configure their pipeline specifically to account for the `virtual:proxysettings` alias which sounds annoying.

Can't we use `import.meta.env` or something along these lines to pass the configuration?

Reply:
Just gave this a test…
Writing an env var during plugin initialization works.
![image](https://user-images.githubusercontent.com/27686785/219873748-cb4830d0-b639-493f-a079-73b2b1ae91ac.png)


There is also the option to add the string into the config object, but I find that to be a bad solution as Vite just replaces all occurrences of the String, regardless of Context (e.g. variables names as the key get expanded to the value; but expansion also happens when inside a string) 

Only downside: The values can only be Strings. But that is not an issue for this use case

Reply:
so core could have an import like 

```ts
const proxyEnabled = import.meta?.env?.VITE_MC_ENABLE_PROXY ?? "false" !== "false"
``` 

No virtual module, and a safe fallback if the context doesnt  "know" about the env var